owner.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * The MIT License (MIT)
+ *
+ * raw_pointers_safe_deletion_tests
+ *
+ * Copyright (C) 2024 Stephen G. Tuggy
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef RAW_POINTERS_SAFE_DELETION_TESTS_OWNER_H
+#define RAW_POINTERS_SAFE_DELETION_TESTS_OWNER_H
+
+#include <memory>
+#include <type_traits>
+
+using std::shared_ptr;
+using std::unique_ptr;
+
+template <class T, class = std::enable_if_t<std::is_pointer<T>::value>>
```

Comment:
Here, I wrote my own `owner` type, so that we don't have to bring in the entire GSL library to make use of it. This general technique is recommended in the Cpp Core Guidelines, I noticed. Probably with good reason. It's important to know when you need to `delete` a pointer, and when you should leave it alone.

What do you guys think about starting to use this? It should not present any compatibility problems with existing code, AFAIK. (Other than, perhaps, requiring C++14, that is. 😄 )